tutorials/migrating-rails-app-from-heroku-to-google-appengine.md

```diff
@@ -6,207 +6,206 @@ tags: App Engine, Ruby, Rails, Heroku, Migration
+    a Cloud SQL instance with the newer version.
 
-2.  Get the connection name for your new instance, as you will need it later.
-    The connection name is a string that looks something like
-    `${PROJECT_ID}:us-central1:mydbinstance`. You can find it by running:
+1.  Get the _connection name_ for your Cloud SQL instance by running the
+    following command:
 
         gcloud sql instances describe mydbinstance
 
-    and looking at the `connectionName` field. We will refer to the connection
-    name as `${DB_CONNECTION_NAME}` when we need it later in this tutorial.
+    In the output, look for the connection name in the `connectionName` field.
+    The connection name has this format: `[PROJECT-ID]:[COMPUTE-ZONE]:mydbinstance` We
+    refer to the connection name as `[CONNECTION-NAME]` throughout this tutorial.
 
-3.  Secure your new database instance by setting a password on the default
-    postgres user. Execute this command and type a password at the prompt:
+1.  Secure your new database instance by setting a password on the default
+    postgres user:
 
         gcloud sql users set-password postgres no-host \
           --instance=mydbinstance --prompt-for-password
+          
+      When prompted, enter a password for the database.
 
-4.  Create a database:
+1.  Create a database:
 
         gcloud sql databases create mydb --instance=mydbinstance
 
-For now your database will be owned by the default user `postgres`. You can
+Currently, your database is owned by the default user, `postgres`. You can
 change the ownership to an unprivileged user later, but for now you should
-keep the default to ensure the data migration steps go smoothly.
+keep the default user to ensure that the data migration goes smoothly.
 
 ### Connect to your Cloud SQL instance
 
-By default, Google Cloud SQL instances are locked down, requiring your IP
-address to be whitelisted in order to connect using the standard Postgres
-protocols. Although this is a security best practice, it does make it difficult
-to establish _ad hoc_ database connections.
-To mitigate this problem, Cloud SQL provides a tool called the
-[Cloud SQL Proxy](https://cloud.google.com/sql/docs/postgres/sql-proxy). It
-opens a database endpoint that you can connect to locally, and then proxies
-back to the database instance via a secure API.
+By default, Cloud SQL instances are secured: to connect using standard
+Postgres protocols, you are required to whitelist your your IP address.
+This security measure, while necessary, makes it challenging to establish
+_ad hoc_ database connections.
 
-You will now set up a Cloud SQL Proxy locally so you can connect to your new
-database.
+Cloud SQL provides the [Cloud SQL Proxy](https://cloud.google.com/sql/docs/postgres/sql-proxy)
+command-line tool, which opens a database endpoint that you can connect to locally and which
+proxies back to the database instance via a secure API.
 
-1.  Enable the Google Cloud SQL API in the cloud console
```

Comment:
@ryanmcginnis I believe we need to enable this API explicitly. This is the SQL admin API (required by the cloud sql proxy), which is not enabled by default.

Reply:
Doesn't look like the SQL Admin API is publicly available. Please confirm.

Reply:
The API is publicly available. The link in my original text (https://console.cloud.google.com/apis/library/sqladmin.googleapis.com/) is accessible externally. (The API names are unfortunately very confusing: the official name is "Google Cloud SQL API" as I had it in the original text, but the symbolic name is "sqladmin".) This API is required in order for the cloud sql proxy to work externally.

Reply:
Gotcha. Thanks for clarifying. In that case, add this step back in:

1. [Enable the Google Cloud SQL API](https://console.cloud.google.com/apis/library/sqladmin.googleapis.com/).

Reply:
Er, here, easier to copy:

1. [Enable the Google Cloud SQL API.] ( https://console.cloud.google.com/apis/library/sqladmin.googleapis.com/ )

Reply:
Done